components/pagination/pagination.ts

```diff
@@ -0,0 +1,166 @@
+import { Component, Input, Output, OnChanges, SimpleChanges, EventEmitter, HostBinding } from "@angular/core";
+
+@Component({
+    selector: "sui-pagination",
+    template: `
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(1)" [class.disabled]="page===1">
+    <span><i class="angle double left icon"></i></span>
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page-1)" [class.disabled]="!hasPrevious()">
+    <span><i class="angle left icon"></i></span>
+</a>
+<a *ngFor="let p of pages" class="item" [class.active]="p===page" (click)="setPage(p)">
+    {{ p }}
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page+1)" [class.disabled]="!hasNext()">
+    <span><i class="angle right icon"></i></span>
+</a>
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(pageCount)" [class.disabled]="page===pageCount">
+    <span><i class="angle double right icon"></i></span>
+</a>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    @HostBinding("class.ui")
+    @HostBinding("class.pagination")
+    @HostBinding("class.menu")
+    private _paginationClasses:boolean;
+
+    // Public members
+    public pageCount:number;
+
+    @Output()
+    public pageChange:EventEmitter<number>;
+
+    // Private members
+    private _maxSize?:number;
```

Comment:
I liked your suggestion of using `undefined` for unlimited pages, however elsewhere in the project I like to use `null` rather than `undefined`. I'd also like to turn on strict null checks soon, so can we make this

```typescript
private _maxSize:number | null;
```